scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -653,13 +653,43 @@ class ScioContext private[scio] (
   /**
    * Runs the underlying pipeline according to the options used to create this context.
    * Running closes the context and no operation can be performed once the context is closed.
+   *
+   * This operation is non-blocking and no operation can be performed once the context is closed.
+   *
+   * @return the [[ScioExecutionContext]] for the underlying job execution.
    */
   def run(): ScioExecutionContext = run(scioRunner)
 
+  /**
+   * Runs the underlying pipeline according to the options used to create this context.
+   * Running closes the context and no operation can be performed once the context is closed.
+   *
+   * This operation is non-blocking and no operation can be performed once the context is closed.
+   *
+   * @param runner [[ScioRunner]] responsible for the execution for this context.
+   *
+   * @return the [[ScioExecutionContext]] for the underlying job execution.
+   */
   def run(runner: ScioRunner): ScioExecutionContext = runner.run(this)
```

Comment:
Add something on how this is different from the default `run()`, like explain the purpose of the runner?

Reply:
I plan to remove this in the stable release ... so I'll not bother with it 😄